tests/sles4sap/hana_cluster.pm

```diff
@@ -60,10 +65,36 @@ sub run {
         barrier_wait "HANA_INIT_CONF_$cluster_name";
         barrier_wait "HANA_CREATED_CONF_$cluster_name";
 
-        # Upload the configuration into the cluster
-        my @crm_cmds = ("crm configure load update $cluster_conf",
-            "crm resource refresh msl_SAPHana_${sid}_HDB${instance_id}",
-            "crm resource maintenance msl_SAPHana_${sid}_HDB${instance_id} off");
+        # Check for ANGI setup
+        if (check_var('ANGI', 'true')) {
+            assert_script_run "su - $sapadm -c 'sapcontrol -nr $instance_id -function StopSystem'";
+            # Setting up SAP HANA HA/DR providers
+            my $hadr_template = "angi_susHanaHADR_AIO.template";
+            assert_script_run "curl -f -v " . autoinst_url . "/data/sles4sap/$hadr_template -o /tmp/$hadr_template";
+            assert_script_run "su - $sapadm -c 'SAPHanaSR-manageProvider --sid $sid --add /tmp/$hadr_template'";
+            # Adding SUDO permissions
+            my $sudo_saphanasr = "# SAPHanaSR-ScaleUp entries for writing srHook cluster attribute and SAPHanaSR-hookHelper\n" .
+              "$sapadm ALL=(ALL) NOPASSWD: /usr/sbin/crm_attribute -n hana_" . lc("$sid") . "_*\n" .
+              "$sapadm ALL=(ALL) NOPASSWD: /usr/bin/SAPHanaSR-hookHelper --sid=" . uc("$sid") . " *\n";
+            write_sut_file("/tmp/etc_sudoers_SAPHanaSR_node_1_$sid", "$sudo_saphanasr");
+            assert_script_run "cp /tmp/etc_sudoers_SAPHanaSR_node_1_$sid /etc/sudoers.d/SAPHanaSR_$sid";
+            my $start_cmd = "su - $sapadm -c 'sapcontrol -nr $instance_id -function StartSystem HDB'";
+            assert_script_run $start_cmd;
+        }
+        # Commits configuration changes into the cluster
+        my @crm_cmds;
+        if (check_var('ANGI', 'true')) {
+            @crm_cmds = ("crm configure load update $cluster_conf",
+                "crm resource refresh mst_SAPHanaCtl_${sid}_HDB${instance_id}",
+                "cs_wait_for_idle -s 5",
+                "crm resource maintenance mst_SAPHanaCtl_${sid}_HDB${instance_id} off");
+        }
+        else {
+            @crm_cmds = ("crm configure load update $cluster_conf",
+                "crm resource refresh msl_SAPHana_${sid}_HDB${instance_id}",
+                "cs_wait_for_idle -s 5",
```

Comment:
You can use single quotes, there is no interpolation required.

Reply:
It was left as it was in original file, but OFC I could change it and make a quotes cocktail everywhere :-)

Reply:
https://perldoc.perl.org/perlglossary#interpolation
If you use double quotes, perl will try to find patterns that can be interpolated. 
Double quotes are used if you want perl to for example replace a variable within a string. If there is no such thing, you use single quotes to avoid accidental interpolation.

```
my $variable = 'variable_value';
print "\n";
print 'Single quoted: $variable';
print "\nDouble quoted: $variable";
print "\n" . "Unintentionally interpolated >>$tring<< with a dollar sign";
print "\n" . 'Unintentionally interpolated >>$tring<< with a dollar sign' . "\n";
```


```
perl ./test.pl 
Single quoted: $variable
Double quoted: variable_value
Unintentionally interpolated >><< with a dollar sign
Unintentionally interpolated >>$tring<< with a dollar sign
```